pallets/parachain-staking/src/auto_compounding.rs

```diff
@@ -37,43 +38,43 @@ pub struct DelegationAutoCompoundConfig<AccountId> {
 	pub value: Percent,
 }
 
-/// Sets the auto-compounding value for a delegation.
-pub fn set_delegation_config<AccountId: Eq>(
+/// Sets the auto-compounding value for a delegation. The `delegations_config` must be a sorted
+/// vector for binary_search to work.
+pub fn set_delegation_config<AccountId: Eq + Ord>(
 	delegations_config: &mut Vec<DelegationAutoCompoundConfig<AccountId>>,
 	delegator: AccountId,
 	value: Percent,
 ) {
-	let maybe_delegation = delegations_config
-		.iter_mut()
-		.find(|entry| entry.delegator == delegator);
-
-	let mut delegation = if let Some(delegation) = maybe_delegation {
-		delegation
-	} else {
-		delegations_config.push(DelegationAutoCompoundConfig {
-			delegator,
-			value: Percent::zero(),
-		});
-		delegations_config.last_mut().expect("cannot fail; qed")
-	};
+	let mut delegation = match delegations_config.binary_search_by(|d| d.delegator.cmp(&delegator)) {
+        Ok(index) => {
+			&mut delegations_config[index]
+		},
+        Err(index) => {
+            delegations_config.insert(index, DelegationAutoCompoundConfig {
+				delegator,
+				value: Percent::zero(),
+			});
+            delegations_config.last_mut().expect("cannot fail; qed")
```

Comment:
Please refactor to remove this expect, the whole function body can be rewritten like this:

```rust
	match delegations_config.binary_search_by(|delegation| delegator.cmp(&delegation.delegator)) {
		Ok(index) => {
			delegations_config[index].value = value;
		}
		Err(index) => {
			delegations_config.insert(index, DelegationAutoCompoundConfig { delegator, value });
		}
	}
```